docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1735,182 @@ As a response to blocked request AdGuard returns a short video placeholder.
+
+The base priority weight of any rule is 1. If the priority is a floating point number, it will be rounded to the **nearest whole number**.
+
+> **Compatibility with different versions of AdGuard**
+>
+> Numerical priorities for rules are available in tsurlfilter since version `v2.1.0`.
+
+
+#### 1. Basic modifiers, the presence of each adds 1 to the priority: {#priority-category-1}
+ * [`$third-party`](#third-party-modifier),
+ * [`$match-case`](#match-case-modifier),
+ * [`$to`](#to-modifier),
+ * [`$method`](#method-modifier),
+ * [`$denyallow`](#denyallow-modifier),
+ * [`$header`](#header-modifier),
+ * [`$dnsrewrite`](#dnsrewrite-modifier),
+ * [`$app`](#app-modifier) with forbidden applications using `~`,
+ * [`$domain`](#domain-modifier) with forbidden domains using `~`,
+ * banned content-types with `~`.
+
+For forbidden `$domain` and content-type, we add 1 for the presence of the modifier itself disregarding the number of forbidden domains or content-types, because the scope of the rule is infinitely large in the limit anyway, in other words, by banning several domains and content-types, we only *insignificantly* narrowed the scope of the rule.
+
+#### 2. Defined content types, $popup, special exceptions: {#priority-category-2}
+* [`$document`](#document-modifier),
+* [`$font`](#font-modifier),
+* [`$image`](#image-modifier),
+* [`$media`](#media-modifier),
+* [`$object`](#object-modifier),
+* [`$other`](#other-modifier),
+* [`$ping`](#ping-modifier),
+* [`$script`](#script-modifier),
+* [`$stylesheet`](#stylesheet-modifier),
+* [`$subdocument`](#subdocument-modifier),
+* [`$websocket`](#websocket-modifier),
+* [`$xmlhttprequest`](#xmlhttprequest-modifier);
+
+This also includes rules that implicitly add the modifier `$document`:
+* [`$popup`](#popup-modifier);
+
+or special exceptions that implicitly add `$document,subdocument`:
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+
+Specified content-types add `50 + 50 / number_of_forbidden_content_types`, for example:
+`||example.com^$image,script` will add `50 + 50 / 2 = 50 + 25 = 75` to the total weight of the rule. The `$popup` also belongs to this category, because it implicitly adds the modifier `$document`. Similarly, specific exceptions add `$document,subdocument`.
+
+#### 3. $domain or $app with allowed domains or applications: {#priority-category-3}
+Specified domains through `$domain` and specified applications through `$app` add `101 + 100 / number_domains (or number_applications)`, for example:
```

Comment:
Not clear what to do if both `$domain` and `$app` modifiers are present. Do we add `101 + 100/(N +M)` (where N and M are the number of included app names and domains respectively), or do we add `202 + 100/N + 100/M`?